app/controllers/sequences_controller.rb

```diff
@@ -179,8 +184,8 @@ def set_sequence
   end
 
   def find_or_create_sequence_run
-    if sequence_run_id = params['sequence_run']
-      @sequence_run = SequenceRun.find(sequence_run_id)
+    if sequence_run_key = params['sequence_run_key']
+      @sequence_run = SequenceRun.find_by_key(sequence_run_key)
       return @sequence_run if @sequence_run
     end
 
```

Comment:
Now that we have sequence run keys, the line below this should be removed. If there isn't a current_user then we should create an anonymous sequence_run.  So the code below needs to be changed to do that creation of the anonymous sequence run.